src/WingetCreateCore/Common/GitHub.cs

```diff
@@ -483,8 +486,23 @@ private async Task UpdateForkedRepoWithUpstreamCommits(Repository forkedRepo)
                     throw new NonFastForwardException(commitsAheadBy);
                 }
 
-                var upstreamBranchReference = await this.github.Git.Reference.Get(upstream.Id, $"heads/{upstream.DefaultBranch}");
-                await this.github.Git.Reference.Update(forkedRepo.Id, $"heads/{forkedRepo.DefaultBranch}", new ReferenceUpdate(upstreamBranchReference.Object.Sha));
+                // Octokit .NET doesn't support sync fork endpoint, so we make a direct call to the GitHub API.
+                // https://docs.github.com/en/rest/branches/branches?apiVersion=2022-11-28#sync-a-fork-branch-with-the-upstream-repository
+                HttpClient httpClient = new HttpClient();
+
+                // Headers
+                httpClient.DefaultRequestHeaders.Authorization = new System.Net.Http.Headers.AuthenticationHeaderValue("Bearer", this.github.Credentials.Password);
+                httpClient.DefaultRequestHeaders.Add("Accept", "application/vnd.github+json");
+                httpClient.DefaultRequestHeaders.Add("X-GitHub-Api-Version", "2022-11-28");
+                httpClient.DefaultRequestHeaders.UserAgent.TryParseAdd(Constants.ProgramName);
+
+                // Payload
+                JsonObject jsonObject = new JsonObject { { "branch", forkedRepo.DefaultBranch } };
+                var content = new StringContent(jsonObject.ToString(), Encoding.UTF8, "application/json");
+
+                var url = $"https://api.github.com/repos/{forkedRepo.Owner.Login}/{forkedRepo.Name}/merge-upstream";
+                var response = await httpClient.PostAsync(url, content);
+                response.EnsureSuccessStatusCode();

```

Comment:
Since the API provides a specific status code for a merge conflict, should there be special handling for the error status codes? Right now this will simply throw the error and exit, but it could be nice to know that `The forked repository could not be synced due to a merge conflict on the default branch` if the response code is `409`

Reply:
We can and I thought of emitting custom exceptions.. but given the blast radius here I didn't want to introduce any resource strings in this PR that could delay getting this checked in as a hotfix.

Reply:
Fair enough; No complaints from me

Reply:
And also, I'm guessing a merge conflict would only ever arise if the fork is ahead of the upstream branch, and for that case the flow would never reach here since it's handled above in

https://github.com/microsoft/winget-create/blob/60cc6f2a8e1613574043cab64fe8767e91dd369d/src/WingetCreateCore/Common/GitHub.cs#L480-L484

Reply:
Is this ok to resolve?

Reply:
Yep! Okay from me @AmelBawa-msft